### PR TITLE
PORTALS-2838, PORTALS-2837: Add Close button to Access Requirement List, and fix Close button for step in Challenge Wizard

### PR DESCRIPTION
**Pull request by jay-hodgson**

### CODE REVIEW

File: packages/synapse-react-client/src/components/ChallengeTeamWizard/ChallengeTeamWizard.tsx [hunk 1]
```diff
@@ -466,9 +466,16 @@ const ChallengeTeamWizard: React.FunctionComponent<
     onClose()
   }
 
-  const onConfirmHandlerMap: Record<string, () => Promise<void>> | void = {
+  const onConfirmHandlerMap: Record<
+    string,
+    (() => Promise<void>) | (() => undefined)
```
Comment (jay-hodgson) [inline packages/synapse-react-client/src/components/ChallengeTeamWizard/ChallengeTeamWizard.tsx:471]: Downstream expects one of these signatures, so I'll just return undefined.

**Comment by jay-hodgson** [comment]
In addition to the modal close button in the upper left, add a Close button in the Dialog actions:
Screen Shot 2023-10-27 at 2 26 52 PM

File: packages/synapse-react-client/src/components/ChallengeTeamWizard/ChallengeTeamWizard.tsx [hunk 2]
```diff
@@ -466,9 +466,16 @@ const ChallengeTeamWizard: React.FunctionComponent<
...
+  > | void = {
     CREATE_NEW_TEAM: handleCreateTeam,
     JOIN_REQUEST_FORM: handleRequestMembership,
+    JOIN_REQUEST_SENT: () => {
```
Comment (jay-hodgson) [inline packages/synapse-react-client/src/components/ChallengeTeamWizard/ChallengeTeamWizard.tsx:475]: The JOIN_REQUEST_SENT didn't have a confirmation handler.  It should just hide the dialog.

File: packages/synapse-react-client/src/components/ChallengeTeamWizard/ChallengeTeamWizard.tsx (hunk 1 above)
Comment (nickgros) [reply packages/synapse-react-client/src/components/ChallengeTeamWizard/ChallengeTeamWizard.tsx:471]: You could just have `() => Promise<void>` and `return Promise.resolve()`, if you think that might make it more simple.

**Review by nickgros — APPROVED** [review]

**Pull request merged**